torch/nn/modules/activation.py

```diff
@@ -1062,6 +1061,11 @@ def forward(self, query: Tensor, key: Tensor, value: Tensor, key_padding_mask: O
             `batch_first` argument is ignored for unbatched inputs.
         """
         is_batched = query.dim() == 3
+        if key_padding_mask is not None:
+            _skpm_dtype = key_padding_mask.dtype
+            if _skpm_dtype != torch.bool and not torch.is_floating_point(key_padding_mask):
```

Comment:
Why call a function rather than query the property directly?
```suggestion
            if _skpm_dtype != torch.bool and not _skpm_dtype.is_floating_point:
```

Reply:
@malfet I tagged you in phabricator. It is because some scripted tensors have really weird tensor dtype, like an "int" instead of torch.int. I pasted some links in pharicator, and msged you directly ;)